src/Agent.Plugins/PipelineArtifact/Telemetry/PipelineTelemetryRecord.cs

```diff
@@ -0,0 +1,25 @@
+using System;
+using Agent.Sdk;
+using Microsoft.VisualStudio.Services.Content.Common.Telemetry;
+using Microsoft.VisualStudio.Services.BlobStore.Common.Telemetry;
+
+namespace Agent.Plugins.PipelineArtifact.Telemetry
+{
+    /// <summary>
+    /// Generic telemetry record for use with Pipeline events.
+    /// </summary>
+    public abstract class PipelineTelemetryRecord : BlobStoreTelemetryRecord
+    {
+        public Guid PlanId { get; private set; }
+        public Guid JobId { get; private set; }
+        public Guid TaskInstanceId { get; private set; }
+
+        public PipelineTelemetryRecord(TelemetryInformationLevel level, Uri baseAddress, string eventNamePrefix, string eventNameSuffix, AgentTaskPluginExecutionContext context, uint attemptNumber = 1)
+            : base(level, baseAddress, eventNamePrefix, eventNameSuffix, attemptNumber)
+        {
+            PlanId = Guid.Parse(context.Variables["system.planId"].Value);
```

Comment:
>Parse [](start = 26, length = 5)

handle invalid guids

Reply:
These are set the system and can't be overridden by the user so they should always be safe. If these values are malformed then something is seriously wrong. Stored as guids on [TaskActivityData](https://dev.azure.com/mseng/AzureDevOps/_git/AzureDevOps?path=%2FReleaseManagement%2FService%2FReleaseManagement2%2FData%2FModel%2FTaskActivityData.cs&version=GBmaster) and are some of the [WellKnownDistributedTaskVariables](https://dev.azure.com/mseng/AzureDevOps/_git/AzureDevOps?path=%2FDistributedTask%2FClient%2FWebApi%2FWebApi%2FWellKnownDistributedTaskVariables.cs&version=GBmaster)

Reply:
Can we use the const so that if it's changed this won't regress

---
In reply to: [289595669](https://github.com/microsoft/azure-pipelines-agent/pull/2274#discussion_r289595669) [](ancestors = 289595669)